tasks/tiler/do.py

```diff
@@ -0,0 +1,122 @@
+            session = current_session()
+            session.execute('CREATE SCHEMA IF NOT EXISTS tiler')
+            cols_schema = []
+            table_name = "{}.{}".format(table_schema['schema'], table_schema['table_name'])
+            for _, cols in table_schema['columns'].items():
+                cols_schema += cols
+            sql_table = '''CREATE TABLE IF NOT EXISTS {}(
+                    x INTEGER NOT NULL,
+                    y INTEGER NOT NULL,
+                    z INTEGER NOT NULL,
+                    mvt_geometry Geometry NOT NULL,
+                    geoid VARCHAR NOT NULL,
+                    area_ratio NUMERIC,
+                    {},
+                    CONSTRAINT xyzusall_pk PRIMARY KEY (x,y,z,geoid)
+                )'''.format(table_name, ", ".join(cols_schema))
+            session.execute(sql_table)
+            session.commit()
+
+    def _get_config_data(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        with (open('{}/us_all.json'.format(dir_path))) as f:
+            return json.load(f)
+
+    def _get_table_schema(self, table_config):
+        columns = {}
+        for dataset, col_data in table_config['columns'].items():
+            columns[dataset] = []
+            for column in col_data:
+                nullable = '' if column['nullable'] else 'NOT NULL'
+                columns[dataset].append("{} {} {}".format(column['column_name'], column['type'], nullable))
+        return {"schema": table_config['schema'], "table_name": table_config['table'], "columns": columns}
+
+    def _generate_tiles(self, zoom, table_schema, columns_config):
+        session = current_session()
+        table_name = "{}.{}".format(table_schema['schema'], table_schema['table_name'])
+        do_columns = [column['id'] for column in columns_config['do']]
+        mc_columns = [column['id'] for column in columns_config['mastercard']]
+        recordset = ["mvtdata->>'id' as id"]
+        recordset.append("(mvtdata->>'area_ratio')::numeric as area_ratio")
+        for dataset, columns in columns_config.items():
+            recordset += ["(mvtdata->>'{}')::{} as {}".format(column['column_name'], column['type'], column['column_name']) for column in columns]
+        for x in range(0, (pow(2, zoom) + 1)):
+            for y in range(0, (pow(2, zoom) + 1)):
+                geography = self._get_geography_level(zoom)
+                sql_tile = '''
+                    INSERT INTO {table}
+                    (SELECT {x}, {y}, {z}, mvtgeom, {recordset}
+                    FROM cdb_observatory.OBS_GetMCDOMVT({z},{x},{y},'{geography}',ARRAY['{docols}']::TEXT[],ARRAY['{mccols}']::TEXT[]));'''.format(table=table_name, x=x, y=y, z=zoom, geography=geography, recordset=", ".join(recordset), docols="', '".join(do_columns), mccols="', '".join(mc_columns))
+                session.execute(sql_tile)
+            session.commit()
+
+    def _get_geography_level(self, zoom):
```

Comment:
As the task receives the `zoom` and the `geography` as parameters, it doesn't make much sense getting the geography level from the zoom here. It would be better if we add a dictionary to get the geography id (for example `us.census.tiger.state`) from the geography (`state`)

Reply:
Yep, I left the duplication to be tackle after but change it please :)